docs/usage/network_policies.md

```diff
@@ -90,6 +92,42 @@ annotations:
 
 This automatically allows the network traffic from the API server pods.
 
+## Additional Namespace Coverage in Garden/Seed Cluster
+
+In some cases, garden or seed clusters might run components in dedicated namespaces which are not covered by the controller by default (see list above).
+Still, it might(/should) be desired to also include such "custom namespaces" into the control of the `NetworkPolicy` controllers.
+
+In order to do so, human operators can adapt the component configs of `gardener-operator` or `gardenlet` by providing label selectors for additional namespaces:
+
+```yaml
+controllers:
+  networkPolicy:
+    additionalNamespaceSelectors:
+    - matchLabels:
+        foo: bar
+```
+
+### Communication With `kube-apiserver` For Components In Custom Namespaces
+
+### Egress Traffic
+
+Component running in such custom namespaces might need to initiate the communication with the `kube-apiserver`s of the virtual garden cluster or a shoot cluster.
+In order to achieve this, their custom namespace must be labeled with `networking.gardener.cloud/access-target-apiserver=allowed`.
+This will make the `NetworkPolicy` controllers automatically provisioning the required policies into their namespace.
+
+As a result, the respective component pods just need to be labeled with
+
+- `networking.resources.gardener.cloud/to-garden-virtual-garden-kube-apiserver-tcp-443=allowed` (virtual garden cluster)
+- `networking.resources.gardener.cloud/to-all-shoots-kube-apiserver-tcp-443=allowed` (shoot clusters)
+
+### Ingress Traffic
+
+Components running in such custom namespaces might serve webhook handlers that must be reached by the `kube-apiservers` of the virtual garden cluster or a shoot cluster.
+In order to achieve this, their `Service` must be annotated with `networking.resources.gardener.cloud/from-all-webhook-targets-allowed-ports=<ports>` as well as
```

Comment:
Is the controller replying to the webhook not the target and therefore the entity calling to the webhook rather the source?
If I read `webhook-targets` I associate that with the controllers running the webhooks and not the `api-server` calling the webhooks. However, the webhook service should be called `from` the `api-server`. Therefore, I would rather understand `from-all-webhook-sources-allowed-ports`.
Please let me know if I misunderstood or am just nit-picking.

Reply:
I understand where you are coming from, and naming is always hard/rarely optimal. The alias is `all-webhook-targets` such that the `kube-apiserver` pods can be labeled with `to-all-webhook-targets=allowed`. OTOH, this means that the `Service`s of such targets must have `from-<alias>-allowed-ports`, hence `from-all-webhook-targets-allowed-ports`.
Again, not optimal, but it's the same for the scrape targets. Since the majority of extensions/components is already adapted accordingly, I'd suggest to keep it like this for now.

Reply:
Thanks for the explanation. That is indeed a hard naming problem. I fine with what we have.